packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -96,7 +95,9 @@ abstract class AssembleBase extends FlutterCommand {
   Environment get environment {
     final FlutterProject flutterProject = FlutterProject.current();
     final Environment result = Environment(
-      buildDir: fs.directory(getBuildDirectory()),
+      buildDir: flutterProject.directory
+          .childDirectory('.dart_tool')
+          .childDirectory('flutter_build'),
```

Comment:
Where will e.g. apks end up?

Reply:
apk/iOS artifacts are currently created by xCode/Gradle, which will only consume the output of assemble via re-entrant flutter calls. I'll need to support placing the artifacts where those tools expect for the near feature, which will most likely be in the form out an `--output=path` argument which copies the non-intermediate outputs into a specified directory. I didn't do that in this PR because I won't have a good idea for the UX until I'm further through hooking up iOS or Android

Reply:
sgtm